tensorflow/contrib/ignite/kernels/ignite_dataset_iterator.cc

```diff
@@ -0,0 +1,437 @@
+
+  if (password_.empty())
+    msg_len += 1;
+  else
+    msg_len += 5 + password_.length();
+
+  TF_RETURN_IF_ERROR(client_->WriteInt(msg_len));
+  TF_RETURN_IF_ERROR(client_->WriteByte(1));
+  TF_RETURN_IF_ERROR(client_->WriteShort(protocol_major_version));
+  TF_RETURN_IF_ERROR(client_->WriteShort(protocol_minor_version));
+  TF_RETURN_IF_ERROR(client_->WriteShort(protocol_patch_version));
+  TF_RETURN_IF_ERROR(client_->WriteByte(2));
+  if (username_.empty()) {
+    TF_RETURN_IF_ERROR(client_->WriteByte(null_val));
+  } else {
+    TF_RETURN_IF_ERROR(client_->WriteByte(string_val));
+    TF_RETURN_IF_ERROR(client_->WriteInt(username_.length()));
+    TF_RETURN_IF_ERROR(
+        client_->WriteData((uint8_t*)username_.c_str(), username_.length()));
+  }
+
+  if (password_.empty()) {
+    TF_RETURN_IF_ERROR(client_->WriteByte(null_val));
+  } else {
+    TF_RETURN_IF_ERROR(client_->WriteByte(string_val));
+    TF_RETURN_IF_ERROR(client_->WriteInt(password_.length()));
+    TF_RETURN_IF_ERROR(
+        client_->WriteData((uint8_t*)password_.c_str(), password_.length()));
+  }
+
+  int32_t handshake_res_len;
+  TF_RETURN_IF_ERROR(client_->ReadInt(&handshake_res_len));
+  uint8_t handshake_res;
+  TF_RETURN_IF_ERROR(client_->ReadByte(&handshake_res));
+
+  LOG(INFO) << "Handshake length " << handshake_res_len << ", res "
+            << (int16_t)handshake_res;
+
+  if (handshake_res != 1) {
+    int16_t serv_ver_major;
+    TF_RETURN_IF_ERROR(client_->ReadShort(&serv_ver_major));
+    int16_t serv_ver_minor;
+    TF_RETURN_IF_ERROR(client_->ReadShort(&serv_ver_minor));
+    int16_t serv_ver_patch;
+    TF_RETURN_IF_ERROR(client_->ReadShort(&serv_ver_patch));
+    uint8_t header;
+    TF_RETURN_IF_ERROR(client_->ReadByte(&header));
+
+    if (header == string_val) {
+      int32_t length;
+      TF_RETURN_IF_ERROR(client_->ReadInt(&length));
+      uint8_t* err_msg_c = new uint8_t[length];
+      TF_RETURN_IF_ERROR(client_->ReadData(err_msg_c, length));
+      std::string err_msg((char*)err_msg_c, length);
+      delete[] err_msg_c;
+
+      return errors::Internal("Handshake Error [result=", handshake_res,
+                              ", version=", serv_ver_major, ".", serv_ver_minor,
+                              ".", serv_ver_patch, ", message='", err_msg,
+                              "']");
+    } else if (header == null_val) {
+      return errors::Internal("Handshake Error [result=", handshake_res,
+                              ", version=", serv_ver_major, ".", serv_ver_minor,
+                              ".", serv_ver_patch, "]");
+    } else {
+      return errors::Internal("Handshake Error [result=", handshake_res,
+                              ", version=", serv_ver_major, ".", serv_ver_minor,
+                              ".", serv_ver_patch, "]");
+    }
+  }
+
+  return Status::OK();
+}
+
+Status IgniteDatasetIterator::ScanQuery() {
+  TF_RETURN_IF_ERROR(client_->WriteInt(25));                   // Message length
```

Comment:
Move the magic number 25 to a named constant, and use it consistently in here.

Reply:
I replaced all magic numbers by constants (where it's appropriate).